src/fu-tool.c

```diff
@@ -1314,6 +1314,51 @@ fu_util_monitor (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_fwdump (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	g_autoptr(GPtrArray) firmware_ids = NULL;
+	g_autoptr(GBytes) blob = NULL;
+
+	/* check args */
+	if (g_strv_length (values) != 1) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_ARGS,
+				     "Invalid arguments: filename expected");
+		return FALSE;
+	}
+
+	/* load file */
+	blob = fu_common_get_contents_bytes (values[0], error);
+	if (blob == NULL)
+		return FALSE;
+
+	/* load engine */
+	if (!fu_engine_load (priv->engine, FU_ENGINE_LOAD_FLAG_NO_ENUMERATE, error))
+		return FALSE;
+
+	firmware_ids = fu_engine_get_firmware_gtype_ids (priv->engine);
+	for (guint i = 0; i < firmware_ids->len; i++) {
+		const gchar *id = g_ptr_array_index (firmware_ids, i);
+		GType gtype = fu_engine_get_firmware_gtype_by_id (priv->engine, id);
+		g_autofree gchar *str = NULL;
+		g_autoptr(FuFirmware) firmware = NULL;
+		g_autoptr(GError) error_local = NULL;
+
+		firmware = g_object_new (gtype, NULL);
+		if (!fu_firmware_parse (firmware, blob, 0, &error_local)) {
+			g_debug ("failed to parse as %s: %s",
+				 g_type_name (gtype), error_local->message);
+			continue;
+		}
+		str = fu_firmware_to_string (firmware);
+		g_print ("%s", str);
+	}
```

Comment:
So the basic thought being you pass this firmware through all known types and then see how it gets interpreted if parsed by the different parsers?

I think I'd rather it be an extra argument of the type of parser to run it through.  Very few firmware will it make sense to just be parsed by a bunch of different parsers.

Reply:
Furthermore if you run with this idea to only run it through one parser, I think it would be good to add another argument to `fwupdtool` for `get-firmware-parsers` that returns a list of known `GTypes` that the bash-completion could potentially use (similar to how `get-plugins` works).

And if you don't pass it into this function as an argument then there can be a prompt for numeric that lets you pick which parser to use.